utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/numeric.scala

```diff
@@ -57,7 +59,27 @@ trait numeric extends MathUtils with HideToString {
   @Documentation(description = "Parse string to number")
   @GenericType(typingFunction = classOf[ToNumberTypingFunction])
   def toNumber(@ParamName("stringOrNumber") stringOrNumber: Any): java.lang.Number = stringOrNumber match {
-    case s: CharSequence     => new java.math.BigDecimal(s.toString)
+    case s: CharSequence =>
+      val ss                                       = s.toString
+      val tryByte: Try[java.lang.Byte]             = Try(java.lang.Byte.parseByte(ss))
```

Comment:
I think that the better approach will be behaving as java behaves when someone declares literal number - I mean skipping byte, short and float cases. I think that these cases are only for some hard computations when you need to save memory. In most business cases, they introduce more noise than it benefits.